vagrant/provision-util.sh

```diff
@@ -0,0 +1,145 @@
+  local master_name=$1
+  local master_ip=$2
+  local -n node_names=$3
+  local -n node_ips=$4
+
+  # Setup hosts file to support ping by hostname to master
+  os::util::add-to-hosts-file "${master_ip}" "${master_name}"
+
+  # Setup hosts file to support ping by hostname to each node in the cluster
+  for (( i=0; i < ${#node_names[@]}; i++ )); do
+    os::util::add-to-hosts-file "${node_ips[$i]}" "${node_names[$i]}"
+  done
+}
+
+os::util::init-certs() {
+  local openshift_root=$1
+  local network_plugin=$2
+  local master_name=$3
+  local master_ip=$4
+  local -n node_names=$5
+  local -n node_ips=$6
+
+  local server_config_dir=${openshift_root}/openshift.local.config
+  local volumes_dir="/var/lib/openshift.local.volumes"
+  local cert_dir="${server_config_dir}/master"
+
+  echo "Generating certs"
+
+  pushd "${openshift_root}"
+
+  # Master certs
+  /usr/bin/openshift admin ca create-master-certs \
+    --overwrite=false \
+    --cert-dir="${cert_dir}" \
+    --master="https://${master_ip}:8443" \
+    --hostnames="${master_ip},${master_name}"
+
+  # Certs for nodes
+  for (( i=0; i < ${#node_names[@]}; i++ )); do
+    local name=${node_names[$i]}
+    local ip=${node_ips[$i]}
+    /usr/bin/openshift admin create-node-config \
+      --node-dir="${server_config_dir}/node-${name}" \
+      --node="${name}" \
+      --hostnames="${name},${ip}" \
+      --master="https://${master_ip}:8443" \
+      --network-plugin="${network_plugin}" \
+      --node-client-certificate-authority="${cert_dir}/ca.crt" \
+      --certificate-authority="${cert_dir}/ca.crt" \
+      --signer-cert="${cert_dir}/ca.crt" \
+      --signer-key="${cert_dir}/ca.key" \
+      --signer-serial="${cert_dir}/ca.serial.txt" \
+      --volume-dir="${volumes_dir}"
+  done
+
+  popd
+}
+
+# Set up the KUBECONFIG environment variable for use by oc
+os::util::set-oc-env() {
+  local deployed_root=$1
+  local target=$2
+
+  if [ "${deployed_root}" = "/" ]; then
+    deployed_root=""
+  fi
+
+  local path="${deployed_root}/openshift.local.config/master/admin.kubeconfig"
+  echo "export KUBECONFIG=${path}" >> "${target}"
+}
+
+os::util::get-network-plugin() {
+  local plugin=$1
+
+  local subnet_plugin="redhat/openshift-ovs-subnet"
+  local multitenant_plugin="redhat/openshift-ovs-multitenant"
+  local default_plugin="${subnet_plugin}"
+
+  if [ "${plugin}" != "${subnet_plugin}" ] && \
+     [ "${plugin}" != "${multitenant_plugin}" ]; then
+    if [ "${plugin}" != "" ]; then
+        >&2 echo "Invalid network plugin: ${plugin}"
+    fi
+    >&2 echo "Using default network plugin: ${default_plugin}"
+    plugin="${default_plugin}"
+  fi
+  echo "${plugin}"
+}
+
+os::util::install-sdn() {
+  local deployed_root=$1
+
+  # Source scripts from an openshift-sdn repo if present to support
+  # openshift-sdn development.
+  local sdn_root="${deployed_root}/third-party/openshift-sdn"
+  if [ -d "${sdn_root}" ]; then
+    pushd "${sdn_root}"
+      make
+      make "install-dev"
+    popd
```

Comment:
Few issues here:
- We may have changes in openshift-sdn that doesn't have corresponding changes in the origin repo (not yet merged or changes for origin are being worked on)
- make install only copies flat/multi-tenant binaries and setup scripts but openshift binary controls these sdn binaries and it can break when order of args or number of args to sdn binary doesn't match.

We can't support git clone/fetch of openshift-sdn until there is feature sync between Godeps/.../openshift-sdn and plugings/osdn dirs. For now don't support third-party/openshift-sdn